Kernel/Syscalls/write.cpp

```diff
@@ -89,7 +89,8 @@ ErrorOr<FlatPtr> Process::do_write(OpenFileDescription& description, UserOrKerne
                 Thread::current()->send_signal(SIGPIPE, &Process::current());
             return nwritten_or_error.release_error();
         }
-        VERIFY(nwritten_or_error.value() > 0);
+        if (nwritten_or_error.value() == 0)
+            break;
```

Comment:
Why is this fix correct? The commit message doesn't explain what the bug was, or how to trigger it.
Also, please add a test demonstrating the bug being fixed :)

Reply:
Thanks, the easiest way to reproduce it is to install the `openssh` port and run:

```shell
ssh test.rebex.net
```

This will give the Kernel panic saying `ASSERTION FAILED: nwritten_or_error.value() > 0`.